src/gha_runner/clouddeployment.py

```diff
@@ -199,11 +199,13 @@ def wait_until_ready(self, ids: list[str], **kwargs):
     def wait_until_removed(self, ids: list[str], **kwargs):
         ec2 = boto3.client("ec2", self.region_name)
         waiter = ec2.get_waiter("instance_terminated")
-        waiter.wait(InstanceIds=ids)
+
         if kwargs:
             waiter.wait(InstanceIds=ids, WaiterConfig=kwargs)
         else:
-            waiter.wait(InstanceIds=ids)
+            # Use a longer WaiterConfig to allow for GPU to properly terminate
+            waiter_config = {"MaxAttempts": 80}
+            waiter.wait(InstanceIds=ids, WaiterConfig=waiter_config)
```

Comment:
I think I'm agreeing with CodeCov here that this appears to be not tested. But maybe that's intentional because it'll be a hassle to test with the default `WaiterConfig={'MaxAttempts': 80}`?

If so, maybe restructure this section something like:

```python
if kwargs:
    waiter_config = kmarts
else:
    waiter_config = {"MaxAttempts": 80}

waiter.wait(InstanceIds=ids, WaterConfig=waiter_config)
```

That way, the call to `waiter.wait` is always hit, and the only thing we skip in testing is implicitly setting a default config.

Alternative: Set the default earlier so that *something* is always passed in with kwargs, and the `if` statement is no longer needed here. (That's probably better, but maybe more work.)

Reply:
Ah I see. I do have a test for this when we are waiting for an instance to start, but not for when we are shutting it down. I have added a test for this case rather than the code in [9382398](https://github.com/omsf-eco-infra/gha-runner/pull/18/commits/93823987c19e28667d7d568b51a4294365be41f4).

Reply:
@dwhswenson seems everything is looking good with the new changes.